### PR TITLE
ZER-17 Implement simple formula field evaluating

### PR DESCRIPTION
**Pull request by PavloMS**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: spec/models/calculation_spec.rb [hunk 1]
```diff
@@ -17,4 +17,31 @@
       .t("#{LOCAL_PREFIX_CALCULATION}.value.too_short"))
     }
   end
+
+  describe '#result' do
+    it 'Should be valid' do
```
Comment (maksymst) [inline spec/models/calculation_spec.rb:22]: Here and below https://github.com/rubocop/rspec-style-guide#should-in-example-docstrings

File: spec/models/calculation_spec.rb [hunk 2]
```diff
@@ -17,4 +17,31 @@
...
+      calculation = Calculation.new(value: 'P1 + 2')
```
Comment (maksymst) [inline spec/models/calculation_spec.rb:23]: use factory to build Calculation

**Review by maksymst — CHANGES_REQUESTED** [review]

File: spec/models/calculation_spec.rb [hunk 3]
```diff
@@ -17,4 +17,31 @@
       .t("#{LOCAL_PREFIX_CALCULATION}.value.too_short"))
     }
   end
+
+  describe '#result' do
+
+    let(:value) { 'P1 + 2' }
+    let(:parameters) { {P1: 2} }
+    subject(:calculation) { create(:calculation, value: value) }
+    subject(:result) { calculation.result(parameters) }
```
Comment (maksymst) [inline spec/models/calculation_spec.rb:26]:
```suggestion
    subject { calculation.result(parameters) }
```

File: spec/models/calculation_spec.rb [hunk 4]
```diff
@@ -17,4 +17,31 @@
       .t("#{LOCAL_PREFIX_CALCULATION}.value.too_short"))
     }
   end
+
+  describe '#result' do
+
+    let(:value) { 'P1 + 2' }
+    let(:parameters) { {P1: 2} }
+    subject(:calculation) { create(:calculation, value: value) }
```
Comment (maksymst) [inline spec/models/calculation_spec.rb:25]:
```suggestion
    let(:calculation) { create(:calculation, value: value) }
```

File: spec/models/calculation_spec.rb [hunk 5]
```diff
@@ -17,4 +17,31 @@
...
+    subject(:result) { calculation.result(parameters) }
+
+    context 'when pass valid data' do
+      it { expect(result).to eq 4 }
```
Comment (maksymst) [inline spec/models/calculation_spec.rb:29]:
```suggestion
      it { is_expected.to eq 4 }
```

**Review by maksymst — CHANGES_REQUESTED** [review]

File: spec/models/calculation_spec.rb (hunk 5 above)
Comment (PavloMS) [reply spec/models/calculation_spec.rb]: Done

**Review by maksymst — APPROVED** [review]

**Review by ochupa — APPROVED** [review]

**Pull request merged**